### PR TITLE
Simplify the spec text of dialect handling

### PR DESCRIPTION
**Pull request by FrankYFTang**
close https://github.com/tc39/intl-displaynames-v2/issues/23

### CODE REVIEW

**Comment by FrankYFTang** [comment]
ping

File: displaynames.html [hunk 1]
```diff
@@ -149,21 +149,19 @@ <h1>Intl.DisplayNames ( _locales_, _options_ )</h1>
...
           1. <ins>Assert: _dialectFields_ is a Record (see <emu-xref href="#sec-Intl.DisplayNames-internal-slots"></emu-xref>).</ins>
-          1. <ins>Let _types_ be _dialectFields_.[[&lt;_type_&gt;]].</ins>
+          1. <ins>Let _styleFields_ be _dialectFields_.[[&lt;_style_&gt;]].</ins>
         1. <ins>Else,</ins>
-          1. <ins>Let _types_ be _dataLocaleData_.[[types]].</ins>
-        1. Assert: _types_ is a Record (see <emu-xref href="#sec-Intl.DisplayNames-internal-slots"></emu-xref>).
-        1. Let _typeFields_ be _types_.[[&lt;_type_&gt;]].
-        1. Assert: _typeFields_ is a Record (see <emu-xref href="#sec-Intl.DisplayNames-internal-slots"></emu-xref>).
-        1. Let _styleFields_ be _typeFields_.[[&lt;_style_&gt;]].
+          1. <ins>Let _styleFields_ be _typeFields_.[[&lt;_style_&gt;]].</ins>
```
Comment (ryzokuken) [inline displaynames.html:164]: Minor editorial nit, but the `Else` can be avoided here by setting to the value of `typeFields` again...

File: displaynames.html [hunk 2]
```diff
@@ -219,11 +217,10 @@ <h1>Internal slots</h1>
...
-        <li>The value of fields <del>`"language"`, </del>`"region"`, `"script"`, <del>and </del>`"currency"`<ins>, `"calendar"`, `"dateTimeField"`, and `"unit"`</ins> must be a Records which must have fields with the names of one of the valid display name styles: `"narrow"`, `"short"`, and `"long"`.</li>
-        <li><ins>The value of fields `"language"` must be a Records which must have fields with the names of one of the valid display name dialect handlings: `"dialectName"`, and `"standardName"`.</ins></li>
+        <li>[[LocaleData]].[[&lt;_locale_&gt;]] must have a [[types]] field for all locale values _locale_. The value of this field must be a Record, which must have fields with the names of one of the valid display name types: `"language"`, `"region"`, `"script"`, <del>and</del> `"currency"`<ins>, `"calendar"`, `"dateTimeField"`, and `"unit"`</ins>.</li>
+        <li><ins>The value of the field `"language"` must be a Records which must have fields with the names of one of the valid display name dialect handlings: `"dialectName"`, and `"standardName"`.</ins></li>
```
Comment (ryzokuken) [inline displaynames.html:221]:
```suggestion
        <li><ins>The value of the field `"language"` must be a Record which must have fields with the names of one of the valid display name dialect handlings: `"dialectName"`, and `"standardName"`.</ins></li>
```

File: displaynames.html [hunk 3]
```diff
@@ -219,11 +217,10 @@ <h1>Internal slots</h1>
...
         <li><ins>The display name dialect handling fields under display name type `"language"` should contain Records which must have fields with the names of one of the valid display name styles: `"narrow"`, `"short"`, and `"long"`.</ins></li>
+        <li>The value of fields <del>`"language"`, </del>`"region"`, `"script"`, <del>and </del>`"currency"`<ins>, `"calendar"`, `"dateTimeField"`, and `"unit"`</ins> must be a Records which must have fields with the names of one of the valid display name styles: `"narrow"`, `"short"`, and `"long"`.</li>
```
Comment (ryzokuken) [inline displaynames.html:223]:
```suggestion
        <li>The value of fields <del>`"language"`, </del>`"region"`, `"script"`, <del>and </del>`"currency"`<ins>, `"calendar"`, `"dateTimeField"`, and `"unit"`</ins> must be a Record which must have fields with the names of one of the valid display name styles: `"narrow"`, `"short"`, and `"long"`.</li>
```

**Review by ryzokuken — APPROVED** [review]
Minor editorial comments, but otherwise LGTM, thanks!

File: displaynames.html (hunk 1 above)
Comment (FrankYFTang) [reply displaynames.html]: PTAL

**Review by ryzokuken — APPROVED** [review]
Looks perfect, thanks!

**Review by sffc — APPROVED** [review]

**Pull request merged**